x/precisebank/keeper/invariants_integration_test.go

```diff
@@ -0,0 +1,121 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/precisebank/keeper"
+	"github.com/kava-labs/kava/x/precisebank/testutil"
+	"github.com/kava-labs/kava/x/precisebank/types"
+	"github.com/stretchr/testify/suite"
+)
+
+type invariantsIntegrationTestSuite struct {
+	testutil.Suite
+}
+
+func (suite *invariantsIntegrationTestSuite) SetupTest() {
+	suite.Suite.SetupTest()
+}
+
+func TestInvariantsIntegrationTest(t *testing.T) {
+	suite.Run(t, new(invariantsIntegrationTestSuite))
+}
+
+func (suite *invariantsIntegrationTestSuite) FundReserve(amt sdkmath.Int) {
+	coins := sdk.NewCoins(sdk.NewCoin(types.IntegerCoinDenom, amt))
+	err := suite.BankKeeper.MintCoins(suite.Ctx, types.ModuleName, coins)
+	suite.Require().NoError(err)
+}
+
+func (suite *invariantsIntegrationTestSuite) TestReserveBackingFractionalInvariant() {
+	tests := []struct {
+		name       string
+		setupFn    func(ctx sdk.Context, k keeper.Keeper)
+		wantBroken bool
+		wantMsg    string
+	}{
+		{
+			"valid - empty state",
+			func(_ sdk.Context, _ keeper.Keeper) {},
+			false,
+			"",
+		},
+		{
+			"valid - fractional balances, no remainder",
```

Comment:
i know it won't happen on mainnet because all akava will be completely backed by ukava, but is no fractional balances & a nonzero remainder ever a valid state we need to test?

Reply:
Added a test case for it - should be broken state since no fractional balances should be 0 remainder